### PR TITLE
Update to "Process start_places and remove copies"

### PR DESCRIPTION
**Pull request by sebastianbarry**
This PR is meant to be a direct upgrade to https://github.com/e-mission/e-mission-phone/pull/951 pointed at the new `may_2023_point_release` branch

I addressed most of your comments from that older PR in this new commit @shankari

### CODE REVIEW

**Comment by sebastianbarry** [comment]
@shankari here is the new e-mission-phone PR for start_confirmed_place changes

File: www/js/diary/infinite_scroll_list.js [hunk 1]
```diff
@@ -400,8 +424,9 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
     }
 
     $scope.populateBasicClasses = function(tripgj) {
-        tripgj.display_start_time = DiaryHelper.getLocalTimeString(tripgj.start_local_dt || tripgj.enter_local_dt);
-        tripgj.display_date = moment((tripgj.start_ts || tripgj.enter_ts) * 1000).format('ddd DD MMM YYYY');
+        if (tripgj.start_ts || tripgj.enter_ts) 
+          tripgj.display_start_time = DiaryHelper.getLocalTimeString(tripgj.start_local_dt || tripgj.enter_local_dt);
+        tripgj.display_date = moment((tripgj.start_ts || tripgj.enter_ts || tripgj.end_ts || tripgj.exit_ts) * 1000).format('ddd DD MMM YYYY');
```
Comment (shankari) [inline www/js/diary/infinite_scroll_list.js:429]: don't you need curly braces here for the `if`?

**Review by shankari — CHANGES_REQUESTED** [review]
I think you still need one fix. Happy to merge after that is fixed and you finish re-testing

File: www/js/diary/infinite_scroll_list.js (hunk 1 above)
Comment (sebastianbarry) [inline www/js/diary/infinite_scroll_list.js:429]:
```suggestion
        if (tripgj.start_ts || tripgj.enter_ts) {
          tripgj.display_start_time = DiaryHelper.getLocalTimeString(tripgj.start_local_dt || tripgj.enter_local_dt);
        }
        tripgj.display_date = moment((tripgj.start_ts || tripgj.enter_ts || tripgj.end_ts || tripgj.exit_ts) * 1000).format('ddd DD MMM YYYY');
```

**Comment by sebastianbarry** [comment]
Testing:

| | Screenshot | Explanation |
:---:|:---:|:---:
| Issue | Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 09 47 Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 09 50 Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 10 01 | As you can see, when we prefill the survey time for the first place object, the time is incorrect and causes you to not be able to submit without changing the time (it sets the time to whatever the current time is) |
| Solution | Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 51 40
Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 51 45 | Now, when you click it it prefills correctly and the input matcher works for the first place entry (the issue here was that the begin timestamp did not exist for the first place object, so we have to auto-fill in the begin timestamp as the start of the day for the exit timestamp) |

**Comment by sebastianbarry** [comment]
Testing with download from server button:

| | | |
:---:|:---:|:---:
| Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 58 40 first place object displays properly | Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 58 51 first place object receives input properly | Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 59 25 download from server correctly downloads entries and does not created duplicates |
| Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 59 45 new first place object is created properly | Simulator Screen Shot - iPhone 11 - 2023-04-27 at 15 59 53 new first place object accepts input properly | Simulator Screen Shot - iPhone 11 - 2023-04-27 at 16 00 18 place additions remain after a refresh on first place objects |

**Comment by sebastianbarry** [comment]
This change should be good to go @shankari

**Comment by shankari** [comment]
@sebastianbarry did you test by pushing the entries and re-running the pipeline to make sure that the end-to-end flow works?

**Comment by sebastianbarry** [comment]
> did you test by pushing the entries and re-running the pipeline to make sure that the end-to-end flow works?

Testing end trip + sync + pipeline run:

| | |
:---:|:---:
| Simulator Screen Shot - iPhone 11 - 2023-04-27 at 17 12 49 | After `end trip + sync` and then running the pipeline again, additions stay within the first place objects and no time-discrepancies are brought about because of the duplicate places/first place object being used |

**Review by shankari — APPROVED** [review]

**Pull request merged**